engine/src/main/java/org/hibernate/search/engine/backend/types/VectorSimilarity.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.backend.types;
+
+/**
+ * Defines a function to calculate the vector similarity, i.e. distance between two vectors.
+ */
+public enum VectorSimilarity {
```

Comment:
I'm wondering if we should be providing more information on these similarity functions... some of them require vectors to be normalized, and some wouldn't be able to handle a vector of `0` for example.... I'd suspect that people who will change it knows what they are doing and at the same time L2 norm looks like a safe default 😃